src/mainviews/create-job.tsx

```diff
@@ -184,10 +201,15 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
       ) {
         newComputeType = envObj.default_compute_type;
       }
+      const newEnvOutputFormats = outputFormatsForEnvironment(
+        environmentList,
+        target.value
+      )?.map(format => format.name);
       props.handleModelChange({
         ...props.model,
         environment: target.value,
-        computeType: newComputeType
+        computeType: newComputeType,
+        outputFormats: newEnvOutputFormats
```

Comment:
The problem is that the UI is not reflecting the actual state of the model when you change the environment. When selecting a new environment, all output formats are re-enabled in the model, but this is not reflected in the UI.

https://user-images.githubusercontent.com/44106031/197876814-ec6df15c-8528-4149-a827-64655cec2ee0.mov

To remedy this, change the `defaultChecked` prop in `output-format-picker.tsx` to `checked`. `defaultChecked` is only respected at initial render, and should only be used for uncontrolled components.